sw/device/tests/crypto/cryptotest/firmware/edn_fi.c

```diff
@@ -0,0 +1,253 @@
+// - V   = 53c78ac61a0bac9d7d2e92b1e73e3392
+// - ReturnedBits =
+// d1c07cd95af8a7f11012c84ce48bb8cb87189e99d40fccb1771c619bdf82ab2280b1dc2f2581f39164f7ac0c510494b3a43c41b7db17514c87b107ae793e01c5
+
+// Seed material for the EDN KAT instantiate command.
+const dif_edn_seed_material_t kEdnKatSeedMaterialInstantiate = {
+    .len = kEdnKatMaxClen,
+    .data = {0x73bec010, 0x9262474c, 0x16a30f76, 0x531b51de, 0x2ee494e5,
+             0xdfec9db3, 0xcb7a879d, 0x5600419c, 0xca79b0b0, 0xdda33b5c,
+             0xa468649e, 0xdf5d73fa},
+};
+// Seed material for the EDN KAT reseed command.
+const dif_edn_seed_material_t kEdnKatSeedMaterialReseed = {
+    .len = kEdnKatMaxClen,
+    .data = {0x73bec010, 0x9262474c, 0x16a30f76, 0x531b51de, 0x2ee494e5,
+             0xdfec9db3, 0xcb7a879d, 0x5600419c, 0xca79b0b0, 0xdda33b5c,
+             0xa468649e, 0xdf5d73fa},
+};
+// Seed material for the EDN KAT generate command.
+const dif_edn_seed_material_t kEdnKatSeedMaterialGenerate = {
+    .len = 0,
+};
+// Expected output data for the EDN KAT.
+const uint32_t kExpectedOutput[kEdnKatOutputLen] = {
+    0xe48bb8cb, 0x1012c84c, 0x5af8a7f1, 0xd1c07cd9, 0xdf82ab22, 0x771c619b,
+    0xd40fccb1, 0x87189e99, 0x510494b3, 0x64f7ac0c, 0x2581f391, 0x80b1dc2f,
+    0x793e01c5, 0x87b107ae, 0xdb17514c, 0xa43c41b7,
+};
+
+status_t handle_edn_fi_bus_ack(ujson_t *uj) {
+  // Enable entropy complex, CSRNG and EDN so Ibex can get entropy.
+  // Configure entropy in auto_mode to avoid starving the system from entropy,
+  // given that boot mode entropy has a limited number of generated bits.
+  TRY(entropy_testutils_auto_mode_init());
+
+  uint32_t ibex_rnd_data[64];
+
+  // Inject faults during generating and receiving random data.
+  // Goal is to manipulate ACK on bus to trigger that the same
+  // data chunk is transmitted multiple times.
+  sca_set_trigger_high();
+  asm volatile(NOP10);
+  for (size_t it = 0; it < 64; it++) {
+    TRY(rv_core_ibex_testutils_get_rnd_data(&rv_core_ibex, kEdnKatTimeout,
+                                            &ibex_rnd_data[it]));
+  }
+  sca_set_trigger_low();
+
+  // Check if there are any collisions.
+  size_t collisions = 0;
+  for (size_t outer = 0; outer < 64; outer++) {
+    for (size_t inner = 0; inner < 64; inner++) {
+      if (outer != inner) {
+        if (ibex_rnd_data[outer] == ibex_rnd_data[inner]) {
+          collisions++;
+        }
+      }
+    }
+  }
+
+  // Read ERR_STATUS register.
+  dif_rv_core_ibex_error_status_t codes;
+  TRY(dif_rv_core_ibex_get_error_status(&rv_core_ibex, &codes));
+
+  // Send result & ERR_STATUS to host.
+  edn_fi_test_result_t uj_output;
+  uj_output.result = collisions;
+  uj_output.err_status = codes;
+  RESP_OK(ujson_serialize_edn_fi_test_result_t, uj, &uj_output);
+  return OK_STATUS();
+}
+
+status_t handle_edn_fi_bus_data(ujson_t *uj) {
+  dif_edn_auto_params_t edn_params =
+      edn_testutils_auto_params_build(true, /*res_itval=*/0, /*glen_val=*/0);
```

Comment:
In general, any randomization in the configuration should be disabled. Because the randomization will be done using either EDN or an LFSR inside Ibex which is reseeded from EDN, meaning the randomization depends on the previous test (fixed or random seed).